spec/acceptance/cli_spec.rb

```diff
@@ -0,0 +1,88 @@
+require 'spec_helper_acceptance'
+
+describe 'installation of server with cli' do
+  it_behaves_like 'an idempotent resource' do
+    let(:manifest) do
+      <<-PUPPET
+      include pulpcore
```

Comment:
If you want to use certificates for client auth, won't you also need to configure Pulp with SSL certificates so Apache is set up that way? https://github.com/theforeman/puppet-pulpcore/blob/master/manifests/init.pp#L22-L30

Reply:
Maybe? Don't look at me like I'd understand all nuances of this module when used outside fpc :stuck_out_tongue_closed_eyes:

Reply:
In retrospect, no, or "maybe, but…": I'm including the basic example, which does the same request via `curl` and there it works:

https://github.com/theforeman/puppet-pulpcore/blob/3dd9f3796a52b6599279ca409078755aa65780f8/spec/support/acceptance/examples.rb#L57-L81

Reply:
Touche. I don't know then. I will try to poke at it locally.

Reply:
I think this is the problem?

```
python38-pulp-cli.noarch                        0.12.0-2.el8                             @pulpcore 
```

https://github.com/pulp/pulp-cli/blob/e567714d063d2b1449b1d54869acaf11fa56b4e7/docs/CHANGES.md#0130-2021-12-16

This would indicate you need at least 0.13+ in order to specify a CA cert and thus use verify_ssl. I would also ask if we could configure the ca-cert in the TOML file? And if not, use PULP_CA_BUNDLE.

Reply:
`REQUESTS_CA_BUNDLE` *should* work with earlier releases, and `pulp status` fails without setting it with the same error we're seeing now with `user list`

Reply:
This is some kind of race condition. Once I reverse the order of `pulp status` and `pulp user list` the former starts to fail…

Reply:
Anything we can do about this?

Reply:
I have no idea where to start looking, to be honest.

We could limit testing to `pulp user list` (as that ensures proper certificate authentication) and try to fix the underlying issue *after* this PR?

Reply:
Some testing suggests it doesn't read the certificate at all.
```console
# export REQUESTS_CA_BUNDLE=/etc/pulpcore-certs/ca-cert.pem
# strace -o pulp pulp user list 
Error: HTTPSConnectionPool(host='centos8-64-1', port=443): Max retries exceeded with url: /pulp/api/v3/users/?offset=0&limit=25 (Caused by SSLError(SSLCertVerificationError(1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: self signed certificate (_ssl.c:1125)')))
# grep pem pulp
# rm -f ~/.cache/squeezer/https___centos8-64-1_/api.json
# strace -o pulp pulp user list 
Error: {"detail":"Authentication credentials were not provided."}
# grep pem pulp
stat("/etc/pulpcore-certs/ca-cert.pem", {st_mode=S_IFREG|0640, st_size=1155, ...}) = 0
stat("/etc/pulpcore-certs/ca-cert.pem", {st_mode=S_IFREG|0640, st_size=1155, ...}) = 0
openat(AT_FDCWD, "/etc/pulpcore-certs/ca-cert.pem", O_RDONLY) = 4
```

So this does appear to be a bug in pulp-cli: it doesn't read the CA file if the cache is already present.

Also, username/password auth doesn't appear to work in our setup but I'm less worried about that.

Reply:
Manually testing the builds from https://github.com/theforeman/pulpcore-packaging/pull/434 show that 0.14 at least works. I suspect https://github.com/pulp/pulp-cli/commit/65de92aba7f33676490e58e498179ca2ad0c1e6a is the fix.

Reply:
Thanks for the debugging. Packaging is updated and this is now green!